.github/workflows/yocto-build-deploy.yml

```diff
@@ -0,0 +1,1174 @@
+name: "Yocto Build-Test-Deploy"
+
+on:
+  workflow_call:
+    secrets:
+      # API key names should be descriptive telling us what company, what for, where from, what's the key.
+      # Let's say, XCompany wants to have test, then keys could be: XCOMPANY_AUTOKIT_BM_ACCESS_TOKEN
+      AUTOKIT_BALENACLOUD_API_KEY:
+        description: balena API key for Leviathan testing from BalenaCloud
+        required: false
+      AUTOKIT_BALENAMACHINE_API_KEY:
+        description: balena API key for Leviathan testing from balena-os BM instance
+        required: false
+      BALENA_API_KEY:
+        description: balena API key for deploying releases # TODO: Different for staging and prod - add another Secret for staging key
+        required: false
+      BALENAOS_STAGING_TOKEN:
```

Comment:
We will use the GH CLI or API to create the respective secrets on the environments at the repo level rather than have all these secrets for each env.

Reply:
added as comments in the workflow file

Reply:
This secret will be deprecated once we deploy the `BALENA_API_KEY` secret to each environment.